lightning/src/onion_message/messenger.rs

```diff
@@ -340,8 +454,27 @@ impl<ES: Deref, NS: Deref, L: Deref, CMH: Deref> OnionMessageHandler for OnionMe
 				log_info!(self.logger,
 					"Received an onion message with path_id {:02x?} and {} reply_path",
 						path_id, if reply_path.is_some() { "a" } else { "no" });
-				match message {
-					OnionMessageContents::Custom(msg) => self.custom_handler.handle_custom_message(msg),
+
+				let response = match message {
+					OnionMessageContents::Offers(msg) => {
+						self.offers_handler.handle_message(msg)
+							.map(|msg| OnionMessageContents::Offers(msg))
+					},
+					OnionMessageContents::Custom(msg) => {
+						self.custom_handler.handle_custom_message(msg)
+							.map(|msg| OnionMessageContents::Custom(msg))
+					},
+				};
+
+				if let Some(response) = response {
+					match response {
+						OnionMessageContents::Offers(_) => self.respond_with_onion_message(
+							response, path_id, reply_path, &*self.offers_handler
+						),
+						OnionMessageContents::Custom(_) => self.respond_with_onion_message(
+							response, path_id, reply_path, &*self.custom_handler
+						),
```

Comment:
Can someone explain the `&*` here? I've been running into this more lately and haven't quite understood it. Also I notice if I remove the `Sized` restriction on `OMH::Target: OffersMessageHandler + Sized`--which is `self.offers_handler`'s type--this gets a warning and I also don't really understand why. I get basic de/referencing and that implementing the `Sized` trait means having a known constant size at compile time but don't understand it's use/need in this context.

Reply:
> `self.custom_handler` is a type implementing `Deref` where the `Target` implements `CustomOnionMessageHandler`. `respond_with_onion_message` wants a reference to a type implementing `ResponseErrorHandler`, which is a supertrait. It seems just calling `deref()` would have sufficed.

Oh I see, okay. Reading up a bit more on `Deref` made me realize there was more happening than I realized. Although, I would've thought Rust would be able to do the type coercion for the supertrait here, hmph.

> IIUC, since `Deref::Target` is `?Sized` (i.e., optionally sized) and `respond_with_onion_message` takes a reference which is `Sized`, we need to restrict its uses to `Sized` types.

I tried to add `?Sized` on the `EH` type in `respond_with_onion_message` and that worked too lol, so I guess for some reason Rust needs us to tell it if it's going to be optionally sized or sized, but I still don't really get why... It doesn't make a difference to the actual code here, so whatever, but weird.